testutils/src/main/java/com/microsoft/identity/internal/testutils/HttpRequestInterceptor.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.internal.testutils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.common.internal.net.HttpClient;
+import com.microsoft.identity.common.internal.net.HttpResponse;
+import com.microsoft.identity.internal.testutils.shadows.ShadowHttpClient;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+
+/**
+ * Intercepting http requests at runtime when the {@link HttpClient} is shadowed with {@link ShadowHttpClient}.
+ *
+ * @see MockHttpClient#setInterceptor(HttpRequestInterceptor)
+ * @see MockHttpClient#setInterceptor(HttpRequestInterceptor, String)
+ * @see MockHttpClient#setInterceptor(HttpRequestInterceptor, HttpClient.HttpMethod)
+ * @see MockHttpClient#setInterceptor(HttpRequestInterceptor, HttpClient.HttpMethod, String)
+ */
+public interface HttpRequestInterceptor {
+
+    /**
+     * @param httpMethod     - the http method
+     * @param requestUrl     - the request url
+     * @param requestHeaders - the request headers
+     * @param requestContent - the request content
+     * @return - the http response object
+     * @throws IOException - throws an exception when something went wrong during the http request
```

Comment:
remove the dashes `-` these probably will cause problems when javadoc is rendered

Reply:
Oh, I see. Let me remove them.